api/volume_server.go

```diff
@@ -539,6 +554,118 @@ func (vs *VolumeServer) StreamOut(w http.ResponseWriter, req *http.Request) {
 	}
 }
 
+func (vs *VolumeServer) GetStreamP2pUrl(w http.ResponseWriter, req *http.Request) {
+	hLog := vs.logger.Session("get-stream-p2p-url")
+	hLog.Debug("start")
+	defer hLog.Debug("done")
+
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		RespondWithError(w, ErrStreamOutFailed, http.StatusInternalServerError)
```

Comment:
Should the error returned here also be `ErrGetStreamP2pUrlFailed`?
```suggestion
		RespondWithError(w, ErrGetStreamP2pUrlFailed, http.StatusInternalServerError)
```

Reply:
Good catch. Fixed.